applications/views.py

```diff
@@ -105,11 +106,11 @@ def applications_csv(request, city):
 
 
 @organiser_only
-def application_detail(request, city, app_id):
+def application_detail(request, city, app_number):
     """
     Display the details of a single application.
     """
-    application = Application.objects.get(pk=app_id)
+    application = [app for app in Application.objects.filter(number=app_number) if app.city == city][0]
```

Comment:
Here, you can use the database will pick the right entry for you.

Suppose you request /yourcity/applications/1 - the filter query here will return application number 1 for every event that has one, and then your list comprehension will perform another database query for each one of those to check which city it is. So as Django Girls gets bigger and bigger, there will be more and more queries here!

It is usually best to let the database do this work! You can add more parameters to the filter query so you can query by both city and number.

Also, the field you have used in the `city` property above will not get the right result - this is the display name of the city, which is not the same as what you have in the URL. The correct field would be `app.form.page.url` (so you could do `.filter(form__page__url=city)`